Modules/Core/Common/include/itkMath.h

```diff
@@ -762,6 +762,55 @@ GreatestPrimeFactor(unsigned long n);
 ITKCommon_EXPORT unsigned long long
 GreatestPrimeFactor(unsigned long long n);
 
+// C++11 does not guarantee that assert can be used in constexpr
+// functions. This is a work-around for GCC 4.8, 4.9. Originating
+// from Andrzej's C++ blog:
+//  https://akrzemi1.wordpress.com/2017/05/18/asserts-in-constexpr-functions/
+#if defined NDEBUG
+#  define ITK_X_ASSERT(CHECK) void(0)
+#else
+#  define ITK_X_ASSERT(CHECK) ((CHECK) ? void(0) : [] { assert(!#CHECK); }())
+#endif
+
+/**  Returns `a * b`. Numeric overflow triggers a compilation error in
+ * "constexpr context" and a debug assert failure at run-time.
+ */
+template <typename TValueType = std::uintmax_t>
+constexpr TValueType
+UnsignedProduct(const std::uintmax_t a, const std::uintmax_t b) ITK_NOEXCEPT
+{
```

Comment:
It would be nice to (also) check here that `TValueType` is unsigned:

      static_assert(std::is_unsigned<TValueType>::value, "Only supports unsigned return types");